### ISSUE TITLE
reproducing the results on inbreast dataset

### ISSUE THREAD

**Issue by CarrotC** [Holliemin9090/Mammographic-mass-CAD-via-pseudo-color-mammogram-and-Mask-R-CNN#3]
Hi, thanks for your great code. I tried to reproduce your result on INbreast dataset but I couldn't get any similar or even close results. Having run Preprocess_mammo_mask.py, I got 2292 mask images and  299 pseudo color images because I deleted the 44 ones with invalid mask. After generating pseudo color image, I used the last 20% of the images for test with the trained model you uploaded. But I got the result Sensitivity:0.058524, False positive per image:0.65, Dice:0.85299@0.15935. 
I got images from DICOM files by code in https://github.com/danishm/mritopng. And I used code in https://gist.github.com/jendelel/3a8e768a8eb9345d49f2a82d02946122 to process xml files for raw annotation images. 
Is there any other specific preprocessing that we should do?

**Comment by Holliemin9090** [comment]
Hi, have you used the “Preprocess_mammo_mask.py” to normalize and pad the mammograms before generating pseudo-color images as stated in Readme file?

**Comment by CarrotC** [comment]
Sorry to disturb again. That's my fault in generating masks from xml files last time. But this time I do get 410  INbreast images and 418 annotation masks (each one contains one lesion annotation or no lesion) after "Preprocess_mammo_mask.py", and get bad results such as "Sensitivity:0.64286，False positive per image:2. 3049，Dice:0.89429@0. 040849". There are 10 models after running mamo_CAD.py train. I tested two of them and neither can get a good result.

**Comment by Holliemin9090** [comment]
Hi, 
The training process is designed to save the model after each epoch so you will have 10 models for 10 epochs. The data were split into 60% training, 20% validation and 20% testing. We selected the model which had the lowes training+validation loss as the final model. The loss can be exported via tensorboard.

**Comment by puspo1997** [comment]
@CarrotC  
Hi, Can you help me to extract masks from XML files? I used the codes you gave above (https://gist.github.com/jendelel/3a8e768a8eb9345d49f2a82d02946122) to process raw annotations. But can not get it.
I will be thankful if anyone helps me.
